### ISSUE TITLE
colorPickerChange and initialization

### ISSUE THREAD

**Issue by BrainCrumbz** [zefoy/ngx-color-picker#34]
Hi there. Original angular2-color-picker component suffers from an issue that, as of today, is still open:

* https://github.com/Alberplz/angular2-color-picker/issues/37
* https://github.com/Alberplz/angular2-color-picker/issues/78

related to unwanted colorPickerChange event firing during initialization. We also had this issue, and in our local project copy we disabled that. 

Looking at this fork code [here](https://github.com/zefoy/ngx-color-picker/blob/5b9270544ca410778e37d2403a7deec09e481a92/src/lib/color-picker.directive.ts#L80) and also giving it a try, it seems the issue is present here as well.

Is that right? In case, do you have any plan to change this behaviour? Can we reason about a PR?

Regards

**Comment by sconix** [comment]
You are true that it does not make sense, I will get rid of it and make a new release.

**Comment by BrainCrumbz** [comment]
Thanks for quick reply. BTW for some reason unknow to me, that is also causing a strange side effect:

on the same page form where we include colorpicker input, we also have a couple of custom components, some implementing `ControlValueAccessor`, some not. As soon as we add the colorpicker directive, the other custom components stop executing Angular lifecycle hooks, in particular we tested `ngOnInit` and `ngAfterContentInit`. They are constructed, but no initialization take places, so later on errors show up when some of their code runs after user intervention.

We tried editing local `color-picker.directive.js` in `node_modules`, like the following:

```js
if (false && /* basically, skip it */ color !== this.colorPicker) {
    //setTimeout(() => {
    this.colorPickerChange.emit(color);
    this.cdr.detectChanges();
}
```

and problem went away, lifecycle hooks are regularly called again.
